tests/test_code_execution.py

```diff
@@ -227,3 +227,71 @@ def test_few_shots(sandbox_type, code_begin, code_end, code_output_begin, code_o
                     assert (
                         generated_output == code_output_begin + expected_output + code_output_end + '\n\n'
                     ), f"{example_name} few shots are failing"
+
+
+@pytest.mark.parametrize("sandbox_type", ['local', 'piston'])
+def test_lean4_basic_code_execution(sandbox_type):
+    sandbox = _get_sandbox(sandbox_type)
+
+    # Test case for correct basic Lean4 code execution
+    correct_code = """
+    -- Test.lean
+    def add (a b : Nat) : Nat :=
+      a + b
+
+    #eval add 3 4  -- This should print 7
+    """
+    expected_output = "7"
+
+    output, session_id = sandbox.execute_code(correct_code, answer_format="lean")
+
+    # Assertions for the correct code
+    assert session_id == None
+    assert output["process_status"] == 'completed', "Expected the process to complete successfully"
+    assert expected_output in output["stdout"], f"Expected the output to include '{expected_output}'"
```

Comment:
Can we check for the full output instead? Or does it contain a lot of information?

Reply:
The full output is actually just the 7, since lean does not output anything by default if it finishes correctly. 
I will change
`expected_output in output["stdout"]` -> `expected_output == output["stdout"]`